storage/remote/client_test.go

```diff
@@ -74,7 +74,7 @@ func TestStoreHTTPErrorHandling(t *testing.T) {
 			t.Fatal(err)
 		}
 
-		err = c.Store(context.Background(), &prompb.WriteRequest{})
+		err = c.Store(context.Background(), prompb.WriteRequest{})
```

Comment:
Lets keep the top-level proto messages as pointers-to-structs, as `proto.Marshall` expects that.